README.md

```diff
@@ -1,56 +1,75 @@
-- Connection.go: the websocket connection
-- homePage.go: serves the index.html file
-    
-## Define    
-- dataDefinitions.go : defines structures for the used data types
-- errorHelpers.go : defines the error used inside the backend
-- jsonHelpers.go : structures of messages received through the websocket and functions to extract data from jsons
-- helper.go : general useful functions that are not in the go std library (find elem in slice for now)
-- securityHelpers.go : functions to check hashes and signatures
+### actors
+This package implements the actors defined by the PoP project, and their reactions to received messages.
 
+### db 
+This package's purpose is to handle interaction with databases. The databases structure is described in the section "Back-end" components.
 
-# Back-end components
+### websocket 
+This package is to handle interactions with the websocket. It has a hub, which handles the publish-subscribe paradigm, and
+the hub has an actor (either Organizer or Witness), that implements the PoP concept.
 
-## Dependencies
+### define
+This package provides structure for "real world events" and corresponding jsonRPC messages,  methods to convert byte received arrays into theses structures
+and function to check validity of the messages and signatures we received.
+
+### jsonRPC
+This package is not used at all, but is defines the protocol we use to format the messages we send and receive.
+
+### test
+This package is, as its name says it, for the tests.
+
+## Back-end components
+
+### Dependencies
 Currently, the project works with the following libraries:
 * [gorilla/websocket](https://github.com/gorilla/websocket): a websocket package for golang. We chose to use this 
 package because it offers a good API, and it is more complete than the websocket package offered by the standard library.
 * [boltdb](https://github.com/boltdb/bolt): a low-level key-value DB management system. It works by creating Buckets, 
 that we can populate with key-value pairs.
 
-## WebSocket Protocol
-The actual implementation (4 files) currently works as following :
-* the `hub.go` maintains a list of opened connections and "routes" the messages. We will change this so there is no more routing to do in the next update.
-* the `connection.go` is the file that properly handles the websocket connection. When it receives a message from the client, it passes 
-it to the hub.
-* the `index.html` acts like the client that prints messages received from the server and that can send messages to the server.
-* the `homePage.go` just serves the `index.html` file on http requests
-### tweaking the server's properties 
+### network testing 
 If you want to test the websocket protocol the server running on a different machine than the client, you have to set the
 server's IP address in `index.html`  at line 36 (instead of `localhost`). You'll have to put your computer's IP address
 before the ":8080" of `main.go` at line 29.
```

Comment:
It is better to have a separate config file rather than directly editing the code.

Reply:
well this file is not "part of the project". It stands for a "minimalist frontend" that just opens a ws, sends messages through it and prints the received messages